operations.cc

```diff
@@ -0,0 +1,92 @@
+#include <assert.h>
+#include "operations.h"
+#include "ioperandsmemory.h"
+
+namespace PolishCalcComponent {
+
+value_t add(IOperandsMemory* memory){
+    auto b = memory->getOperand();
+    auto a = memory->getOperand();
+
+    memory->putOperand(a + b);
+}
+
+value_t sub(IOperandsMemory* memory){
+    auto b = memory->getOperand();
+    auto a = memory->getOperand();
+
+    memory->putOperand(a - b);
+}
+
+value_t div(IOperandsMemory* memory){
+    auto b = memory->getOperand();
+    auto a = memory->getOperand();
+
+    memory->putOperand(a / b);
+}
+
+value_t mul(IOperandsMemory* memory){
+    auto b = memory->getOperand();
+    auto a = memory->getOperand();
+
+    memory->putOperand(a * b);
+}
+
+value_t neg(IOperandsMemory* memory){
+    auto a = memory->getOperand();
+
+    memory->putOperand( -a );
+}
+
+enum class Operations: char{
+    ADD = '+',
+    SUB = '-',
+    MUL = '*',
+    DIV = '/',
+    NEG = '!'
+};
+
+value_t getValue(const std::string &valStr)
+{
+    std::string::size_type sz;
+    value_t value = std::stod(valStr, &sz);
+    assert(sz == valStr.size());
```

Comment:
I hope, that `getValue` used only on strings that are already validated. In other case, you may have crash in debug (that is good), but silent error in release (that is bad)